### PR TITLE
Allow istios dest rule LB to be nullable

### PR DESCRIPTION
**Pull request by cbron**
Vendoring and integration test for: https://github.com/rancher/types/pull/1142 
Issue: https://github.com/rancher/rancher/issues/25515

### CODE REVIEW

File: tests/integration/suite/test_istio.py [hunk 1]
```diff
@@ -66,6 +66,45 @@ def test_destination_rule(admin_pc):
...
+        }
+    ),
+    destinationRules = client.list_destinationRule(
+        namespaceId=ns.id
+    )
+    assert len(destinationRules) == 1
+    assert destinationRules.data[0].trafficPolicy.loadBalancer\
+        .consistentHash.httpCookie.name == cookie_name
+    client.delete(destinationRules.data[0])
+    client.delete(ns)
```
Comment (rmweir) [inline tests/integration/suite/test_istio.py:105]: Should use remove_resource. In current state, if something fails they won't be deleted.

File: tests/integration/suite/test_istio.py [hunk 2]
```diff
@@ -66,6 +66,45 @@ def test_destination_rule(admin_pc):
     client.delete(ns)
 
 
+@pytest.mark.nonparallel
+def test_destination_rule_on_cookie(admin_pc):
```
Comment (rmweir) [inline tests/integration/suite/test_istio.py:70]: a small summary here to explain what this addresses or why it's being tested would be good

File: tests/integration/suite/test_istio.py [hunk 3]
```diff
@@ -66,6 +66,45 @@ def test_destination_rule(admin_pc):
     client.delete(ns)
 
 
+@pytest.mark.nonparallel
```
Comment (rmweir) [inline tests/integration/suite/test_istio.py:69]: Do we need it to be nonparallel?

Comment (cbron) [reply tests/integration/suite/test_istio.py:69]: Probably not for a destination rule, for virtual services they were likely testing similar things so they could mess with each other. I'll update.

File: tests/integration/suite/test_istio.py [hunk 4]
```diff
@@ -66,6 +66,48 @@ def test_destination_rule(admin_pc):
...
+            "loadBalancer": {
+                "consistentHash": {
+                    "httpCookie": {
+                        "ttl": "0s",
+                        "name": cookie_name,
+                    }
+                }
+            }
+        }
+    ),
```
Comment (rmweir) [inline tests/integration/suite/test_istio.py:101]: nit: remove_resource on destinationRule too, although it should be fine since ns will be deleted

Comment (cbron) [reply tests/integration/suite/test_istio.py]: fixed

File: tests/integration/suite/test_istio.py [hunk 5]
```diff
@@ -66,6 +66,48 @@ def test_destination_rule(admin_pc):
     client.delete(ns)
 
 
+# consistentHash has an "anyOf" openAPI validation on it,
```
Comment (rmweir) [inline tests/integration/suite/test_istio.py:69]: do you mean oneOf? AnyOf would allow both to be filled in.

Comment (cbron) [reply tests/integration/suite/test_istio.py:69]: dang, you're right

Comment (cbron) [reply tests/integration/suite/test_istio.py]: fixed

**Pull request closed without merging**